### PR TITLE
Vfolder data tools

### PR DESCRIPTION
**Pull request by phlax**
Adds data tools to aggregate store data for vfolders

### CODE REVIEW

File: pootle/apps/virtualfolder/utils.py [hunk 1]
```diff
@@ -217,3 +218,39 @@ def update_stores(self):
...
+    def all_stat_data(self):
+        return self.data_model.filter(
+            store__pootle_path__startswith=self.context.pootle_path)
+
+    def vfolder_is_visible(self, vfolder):
+        return (
+            vfolder["store__vfolders__is_public"]
+            and (vfolder["critical"]
+                 or vfolder["suggestions"]
+                 or (vfolder["store__vfolders__priority"] > 1
```
Comment (unho) [inline pootle/apps/virtualfolder/utils.py]: Use `>= 1` as in https://github.com/translate/pootle/blob/master/pootle/apps/virtualfolder/models.py#L161

File: pootle/apps/virtualfolder/utils.py [hunk 2]
```diff
@@ -217,3 +218,41 @@ def update_stores(self):
...
+            user=user, aggregate=False).get("children") or {}
+        vfolders = {
+            vf.name: vf
+            for vf
+            in VirtualFolder.objects.filter(name__in=stats.keys())}
+        for k, v in stats.items():
+            vfolder = vfolders.get(k)
+            stats[k]["priority"] = vfolder.priority
+            stats[k]["isVisible"] = (
+                vfolder.is_public
```
Comment (unho) [inline pootle/apps/virtualfolder/utils.py]: Just wondering why not filter in the queryset above.

File: pootle/apps/virtualfolder/utils.py (hunk 1 above)
Comment (unho) [inline pootle/apps/virtualfolder/utils.py]: Addressed.

**Comment by unho** [comment]
Nit aside, lgtm.

File: pootle/apps/virtualfolder/utils.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/virtualfolder/utils.py]: because admins need it

Comment (unho) [inline pootle/apps/virtualfolder/utils.py]: Sure.

**Pull request merged**